src/Tools/dotnet-monitor/Exceptions/ExceptionsStore.cs

```diff
@@ -104,24 +105,85 @@ private async Task ProcessEntriesAsync(CancellationToken token)
                         moduleName = NameFormatter.GetModuleName(entry.Cache.NameCache, exceptionClassData.ModuleId);
                     }
 
+                    CallStackResultModel callStackResult = GenerateCallStackResult(entry.StackFrameIds, entry.Cache, entry.ThreadId);
+
                     lock (_instances)
                     {
-                        _instances.Add(new ExceptionInstance(exceptionTypeName, moduleName, entry.Message, entry.Timestamp));
+                        _instances.Add(new ExceptionInstance(exceptionTypeName, moduleName, entry.Message, entry.Timestamp, callStackResult));
                     }
                 }
 
                 shouldReadEntry = await _channel.Reader.WaitToReadAsync(token);
             }
         }
 
+        internal static CallStackResultModel GenerateCallStackResult(ulong[] stackFrameIds, IExceptionsNameCache cache, int threadId)
```

Comment:
Why not return [CallStack](https://github.com/dotnet/dotnet-monitor/blob/main/src/Microsoft.Diagnostics.Monitoring.WebApi/Models/CallStackResults.cs#L25) instead of CallStackResult? A thrown exception will only have a single call stack.